### PR TITLE
Document kubelet image credential provider extension concept

### LINKED ISSUE #32048: Document kubelet image credential provider extension concept
**This is a Feature Request**

**What would you like to be added**
Mention kubelet image credential providers within https://kubernetes.io/docs/concepts/extend-kubernetes/

**Why is this needed**
The concepts section about [extending Kubernetes](https://kubernetes.io/docs/concepts/extend-kubernetes/) does not list kubelet credential providers. 
https://kubernetes.io/docs/tasks/kubelet-credential-provider/kubelet-credential-provider/ exists but largely in isolation; you'd have to know it existed to find it.

It's also OK to mention this concept elsewhere - and _especially_ encouraged if the feature looks likely to graduate towards beta.

**Comments**
/sig node
/language en
/kind feature

### PR DESCRIPTION
**Pull request by mrgiles**
Document the kubelet image credential provider extension concept. Fixes #32048

### CODE REVIEW

**Comment by mrgiles** [comment]
FYI @sftim

File: content/en/docs/concepts/extend-kubernetes/_index.md [hunk 1]
```diff
@@ -283,6 +283,23 @@ and to support other aspects of the Kubernetes network model.
 [Network Plugins](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/)
 allow Kubernetes to work with different networking topologies and technologies.
 
+### kubelet image credential provider extension
+
+The kubelet image credential provider extension is a plugin framework that allows the kubelet to dynamically
+fetch credentials for pulling container images from any container image registry using external plugins.
```
Comment (tengqm) [inline content/en/docs/concepts/extend-kubernetes/_index.md:289]: Please avoid long sentences like this.
This sentence is difficult to parse. Is the external plugins used for pulling images or is it used for fetching credentials?

Comment (mrgiles) [reply content/en/docs/concepts/extend-kubernetes/_index.md]: You're right. Sorry.

File: content/en/docs/concepts/extend-kubernetes/_index.md [hunk 2]
```diff
@@ -283,6 +283,23 @@ and to support other aspects of the Kubernetes network model.
 [Network Plugins](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/)
 allow Kubernetes to work with different networking topologies and technologies.
 
+### kubelet image credential provider extension
+
+The kubelet image credential provider extension is a feature that allows the kubelet to dynamically fetch
+credentials for any container image registry using plugins. It was made generally available (GA) in
+Kubernetes v1.26.
```
Comment (tengqm) [inline content/en/docs/concepts/extend-kubernetes/_index.md:290]:
```suggestion
The kubelet image credential providers are plugins for the kubelet to dynamically
retrieve image registry credentials. The credentials are then used when pulling images
from any container image registries. The feature was made generally available
(GA) in Kubernetes v1.26.
```

**Comment by tengqm** [comment]
Please always squash your PR into a single commit. There are currently 3 commits in it.
/approve

**Comment by sftim** [comment]
> Please always squash your PR into a single commit. There are currently 3 commits in it.

Now there are 4. If you're willing to [squash](https://www.kubernetes.dev/docs/guide/pull-requests/#squashing), please do.

If you'd like, as a maintainer I can do that squash for you and push to your fork @mrgiles.

**Comment by sftim** [comment]
/reopen
Misclicked, sorry.

**Comment by mrgiles** [comment]
Sorry @sftim, I'm still struggling with the squashing technique. Is it okay now? 
FYI @reylejano

File: content/en/docs/concepts/extend-kubernetes/_index.md [hunk 3]
```diff
@@ -283,6 +283,19 @@ and to support other aspects of the Kubernetes network model.
 [Network Plugins](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/)
 allow Kubernetes to work with different networking topologies and technologies.
 
+### Kubelet image credential provider plugins
+
+Kubelet image credential providers are plugins for the kubelet to dynamically retrieve image registry
+credentials. The credentials are then used when pulling images from any container image registries.
+The feature was made generally available (GA) in Kubernetes v1.26.
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/_index.md:290]:
```suggestion

{{< feature-state for_k8s_version="v1.26" state="stable" >}}

Kubelet image credential providers are plugins for the kubelet to dynamically retrieve image registry
credentials. The credentials are then used when pulling images from container image registries that
match the configuration.

```

**Review by sftim — COMMENTED** [review]
Try this, which uses a [feature state](https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#feature-state) shortcode. That's what our style guide recommends.

File: content/en/docs/concepts/extend-kubernetes/_index.md (hunk 3 above)
Comment (mrgiles) [reply content/en/docs/concepts/extend-kubernetes/_index.md]: Done. Thanks for the suggestion @sftim

**Review by sftim — COMMENTED** [review]
Thanks 

/lgtm

**Pull request merged**